### ISSUE TITLE
图聚类数目随着worker数量增加而倍增的问题

### ISSUE THREAD

**Issue by xiachaoqun93** [alibaba/euler#229]
你好，我人工构建了一个强假设图结构，包含10个clusters，在用k个worker进行分布式训练时，最后输出的embedding可视化后会产生10*k个clusters。已经调试过每个worker的sample_node和sample_edge可以采样到0-9不同尾号的node，请问这个是哪里的bug啊？

**Comment by alinamimi** [comment]
把你的数据信息，使用的算法，中间训练过程，详细信息贴一下吧

**Comment by xiachaoqun93** [comment]
> 把你的数据信息，使用的算法，中间训练过程，详细信息贴一下吧

你好，图数据的是node_weight =

**Comment by xiachaoqun93** [comment]
> 把你的数据信息，使用的算法，中间训练过程，详细信息贴一下吧

你好，图数据的格式node_type=0, edge_type=0，node_weight和edge_weight恒设定为1；

**Comment by xiachaoqun93** [comment]
感谢回复，图数据的格式node_type=0, edge_type=0，node_weight和edge_weight恒设定为1；
训练日志见附件。
祝好！
[log.txt](https://github.com/alibaba/euler/files/4269851/log.txt)

**Comment by alinamimi** [comment]
图数据分多少份？起几个ps，几个worker？启动命令是什么？

**Comment by xiachaoqun93** [comment]
> 图数据分多少份？起几个ps，几个worker？启动命令是什么？

图数据分为10份，3个ps，10个worker，是挂载到xlearning上启动的，启动命令：
$XLEARNING_HOME/bin/xl-submit \
 --app-type "EULER" \
 --app-name "merchant-graph-train-diy-xcq" \
 --output "/tmp/xcq_euler_model_$RANDOM#model" \
 --launch-cmd "sh xl_run.sh --data_dir $data_path --model_dir $model_path --max_id $max_id --model line --dim 128 --order 2 --optimizer sgd --aggregator mean --num_epochs 500 --learning_rate 0.1 --mode train" \
 --files xl_run.sh,ks_models.tar \
 --ps-num 3 \
 --ps-memory 15G \
 --ps-cores 5 \
 --worker-num 10 \
 --worker-memory 40G \
 --worker-cores 10 \
 --queue root.learn.dp \
 --container-type "docker" \
 --docker-image "arthur/euler-v012" \
 --conf "xlearning.zookeeper.addr"="bjpg-rs320.yz02:2181" \
 --conf "xlearning.zookeeper.path"="/euler"

**Comment by alinamimi** [comment]
xl_run.sh  这个里面的具体命令是什么

**Comment by xiachaoqun93** [comment]
> xl_run.sh 这个里面的具体命令是什么

就是python run_loop.py

**Comment by alinamimi** [comment]
https://github.com/alibaba/euler/wiki/%E5%BF%AB%E9%80%9F%E5%BC%80%E5%A7%8B#%E5%88%86%E5%B8%83%E5%BC%8F%E8%AE%AD%E7%BB%83
你可以参考一下这个，感觉是worker和ps同步之间有问题

**Comment by xiachaoqun93** [comment]
> https://github.com/alibaba/euler/wiki/%E5%BF%AB%E9%80%9F%E5%BC%80%E5%A7%8B#%E5%88%86%E5%B8%83%E5%BC%8F%E8%AE%AD%E7%BB%83
> 你可以参考一下这个，感觉是worker和ps同步之间有问题
感谢大佬花时间解答，我再好好琢磨一下，谢谢！

**Comment by wsnooker** [comment]
> https://github.com/alibaba/euler/wiki/%E5%BF%AB%E9%80%9F%E5%BC%80%E5%A7%8B#%E5%88%86%E5%B8%83%E5%BC%8F%E8%AE%AD%E7%BB%83
> 你可以参考一下这个，感觉是worker和ps同步之间有问题

我理解这里单个worker内聚类效果好，而全局聚类效果变多，例如：数据10类，单个worker聚10类，但2个worker数据聚类20类，是不是和全局参数更新有关，即通过rpc获取的其他worker节点，取不到对应的embedding（或者直接给的默认初始值），没有参与更新，相当于不同worker数据参数没有共享，才导致每个worker参数独立更新，不影响其他节点；
